### ISSUE TITLE
How to interpret the trained model inside models/sup

### ISSUE THREAD

**Issue by Ajay-Ravindran** [Jlfj345wf/RAPO#1]
We have trained the model using train_sup.sh for English to Hindi and obtained the pickle file, containing the DssmTrainer class object. Our requirement is to generate a term in Hindi given a word in English. We are not clear on how to use the pickled file (or the source and target mapping vectors in the DssmTrainer class object) to generate the target term. We tested by taking a word in English and it's corresponding word in Hindi. We tried multiplying the source mapping vectors with the embedding of the English word, and similarly multiplied the target mapping vector with the embedding of the Hindi word. The cosine similarity obtained was pretty low for related words in English and Hindi. Are we missing something?

The code we used for the inference is below:

```import numpy as np
from numpy.linalg import norm
import pickle
import io

def cosine(emb1,emb2):
    return np.dot(np.transpose(emb1),emb2)/(norm(emb1)*norm(emb2))


def load_vectors(fname):
    fin = io.open(fname, 'r', encoding='utf-8', newline='\n', errors='ignore')
    n, d = map(int, fin.readline().split())
    data = {}
    for line in fin:
        tokens = line.rstrip().split(' ')
        data[tokens[0]] = list(map(float, tokens[1:]))
    return data


with open('models/sup/en-hi_model.pkl_0','rb') as f:
    data = pickle.load(f)

src_mapping_vector = np.empty([300,300])
tgt_mapping_vector = np.empty([300,300])

i = 0
for p in data.model.src_tower.mapping_vectors:
    src_mapping_vector[i] = p.data.detach().cpu().numpy()
    i += 1
# src_mapping_vector = np.transpose(src_mapping_vector)

i = 0
for p in data.model.tgt_tower.mapping_vectors:
    tgt_mapping_vector[i] = p.data.detach().cpu().numpy()
    i += 1
# tgt_mapping_vector = np.transpose(tgt_mapping_vector)


en_embs = load_vectors("data/embeddings/wiki.en.vec")
hi_embs = load_vectors("data/embeddings/wiki.hi.vec")

en_emb_word = np.array(en_embs["book"])
hi_emb_word = np.array(hi_embs["किताब"])

en_emb_converted = np.matmul(src_mapping_vector, en_emb_word)
hi_emb_converted = np.matmul(tgt_mapping_vector, hi_emb_word)
print("Similarity:",cosine(en_emb_converted, hi_emb_converted))

**Comment by Jlfj345wf** [comment]
We have responded to your question in an email.
